src/main/java/codeu/controller/ActivityFeedServlet.java

```diff
@@ -1,25 +1,171 @@
+		HashMap<Instant, HashMap<UUID, String>> outerhm = new HashMap<Instant, HashMap<UUID, String>>();
+		if (users != null) {
+		for (User user : users) {
+			HashMap<UUID, String> innerhm = new HashMap<UUID, String>();
+			innerhm.put(user.getId(), "user");
+			outerhm.put(user.getCreationTime(), innerhm);
+		} }
+		if (conversations != null) {
+		for (Conversation conversation : conversations) {
+			HashMap<UUID, String> innerhm = new HashMap<UUID, String>();
+			innerhm.put(conversation.getId(), "conversation");
+			outerhm.put(conversation.getCreationTime(), innerhm);
+		} }
+		if (messages != null) {
+		for (Message message : messages) {
+			HashMap<UUID, String> innerhm = new HashMap<UUID, String>();
+			innerhm.put(message.getId(), "message");
+			outerhm.put(message.getCreationTime(), innerhm);
+		}}
+		return outerhm;
+	}
+
+	/**
+	 * This function takes in a HashMap and finds the earliest Instant using the
+	 * isBefore function
+	 * 
+	 * @param hm
+	 * @return the earliest instant
+	 */
+	Instant findEarliestInstant(HashMap<Instant, HashMap<UUID, String>> hm) {
+		Instant earliestInstant = null;
+		for (Map.Entry<Instant, HashMap<UUID, String>> m : hm.entrySet()) {
+			earliestInstant = m.getKey();
+			break;
+		}
+		for (Map.Entry<Instant, HashMap<UUID, String>> m : hm.entrySet()) {
+			Instant toCmpInstant = m.getKey();
+			if (earliestInstant.isBefore(toCmpInstant)) {
+				earliestInstant = toCmpInstant;
+			}
+		}
+		return earliestInstant;
+	}
+
+	/**
+	 * This function takes in a HashMap and sorts the instants from latest to oldest
+	 * Stores Instants in correct order in an ArrayList
+	 * 
+	 * @param hm
+	 * @return a sorted HashMap by Instants
+	 */
+	HashMap<Instant, HashMap<UUID, String>> sortHashMap(HashMap<Instant, HashMap<UUID, String>> hm) {
+		Instant earlier = null;
+		int size = hm.size();
+		arrInstantsSorted = new ArrayList<Instant>();
+		HashMap<Instant, HashMap<UUID, String>> sortedhm = new HashMap<Instant, HashMap<UUID, String>>();
+		for (int i = 0; i < size; i++) {
+			earlier = findEarliestInstant(hm);
+			arrInstantsSorted.add(earlier);
+			HashMap<UUID, String> innersm = hm.get(earlier);
+			sortedhm.put(earlier, innersm);
+			hm.remove(earlier);
+		}
+		return sortedhm;
+	}
+	
+	/*
+	 * For testing purposes
+	 */
+	ArrayList<Instant> getSortedInstants(){
+		return arrInstantsSorted;
+		
+	}
+
+	/**
+	 * This function fires when a user requests the /activity URL. It simply
+	 * forwards the request to activity.jsp.
+	 * 
+	 * @throws IOException
+	 * @throws ServletException
+	 */
+	@Override
+	public void doGet(HttpServletRequest request, HttpServletResponse response) throws ServletException, IOException {
+		List<User> users = UserStore.getAllUsers();
+		List<Conversation> conversations = ConversationStore.getAllConversations();
+		List<Message> messages = MessageStore.getAllMessages();
+		HashMap<Instant, HashMap<UUID, String>> instantByInstance = buildHashMap(conversations, users, messages);

```

Comment:
What does the variable name instantByInstance mean? I could be missing something, but it kind of confused me.